### PR TITLE
Support all NULL-syntax X.509v3 extensions

### PR DESCRIPTION
**Pull request by JonathanWilbur**
This PR adds support for six X.509v3 extensions that all have the same exact syntax: `NULL`. Most of these are used for attribute certificates, but `sOAIdentifier` is used for public key certificates.

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Comment by JonathanWilbur** [comment]
The extensions are defined in [ITU Recommendation X.509 (2019)](https://www.itu.int/rec/T-REC-X.509-201910-I/en), which is available for free.

File: crypto/x509/v3_group_ac.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 1999-2021 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (paulidale) [inline crypto/x509/v3_group_ac.c:2]: The year looks wrong here.

File: test/certs/ext-groupAC.pem [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+MIIBhjCCAXKgAwIBAgIDAQIDMAsGCSqGSIb3DQEBBTAAMCIYDzIwMjEwODMwMDA1
+MzE3WhgPMjAyMTA4MzAwMDUzMTdaMAAwggEgMAsGCSqGSIb3DQEBAQOCAQ8AMIIB
+CgKCAQEAtnjLm1ts1hC4fNNt3UnQD9y73bDXgioTyWYSI3ca/KNfuTydjFTEYAmq
+nuGrBOUfgbmH3PRQ0AmpqljgWTb3d3K8H4UFvDWQTPSS21IMjm8oqd19nE5GxWir
+Gu0oDRzhWLHe1RZ7ZrohCPg/1Ocsy47QZuK2laFB0rEmrRWBmEYbDl3/wxf5XfqI
+qpOynJB02thXrTCcTM7Rz1FqCFt/ZVZB5hKY2S+CTdE9OIVKlr4WHMfuvUYeOj06
+GkwLFJHNv2tU+tovI3mYRxUuY4UupkS3MC+Otey7XKm1P+INjWWoegm6iCAt3Vus
+pVz+6pU2xgl3nrAVMQHB4fReQPH0pQIDAQABow0wCzAJBgNVHUIEAgUAMAsGCSqG
+SIb3DQEBBQMBAA==
+-----END CERTIFICATE-----
```
Comment (paulidale) [inline test/certs/ext-groupAC.pem:11]: We'd normally have a new line at the end of the file.

**Review by paulidale — CHANGES_REQUESTED** [review]

File: crypto/x509/standard_exts.h [hunk 3]
```diff
@@ -71,7 +72,13 @@ static const X509V3_EXT_METHOD *standard_exts[] = {
     &ossl_v3_utf8_list[0],
     &ossl_v3_issuer_sign_tool,
     &ossl_v3_tls_feature,
-    &ossl_v3_ext_admission
+    &ossl_v3_ext_admission,
+    &ossl_v3_ext_admission,
```
Comment (slontis) [inline crypto/x509/standard_exts.h:76]: Why is ossl_v3_ext_admission duplicated?

File: crypto/x509/v3_group_ac.c (hunk 1 above)
Comment (slontis) [reply crypto/x509/v3_group_ac.c:2]: This is duplicated in most files... Should just be "Copyright 2023 The OpenSSL ..."

File: crypto/x509/v3_ind_iss.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/asn1.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+#include "ext_dat.h"
+
```
Comment (slontis) [inline crypto/x509/v3_ind_iss.c:16]: Could you add the reference as a comment in each of the extension files..
e.g
````
/*
 * See https://...... Section 7.5.2.5 Indirect issuer extension
 */
````

File: crypto/x509/standard_exts.h (hunk 3 above)
Comment (JonathanWilbur) [reply crypto/x509/standard_exts.h]: My mistake. I thought the diff was just the comma added at the end. Fixed.

**Comment by JonathanWilbur** [comment]
All feedback addressed / fixed.

**Comment by JonathanWilbur** [comment]
Dang it, I just realized I didn't sign-off on the last three commits. Is there any sensible way I can do that, or should I just open a new PR?

**Comment by slontis** [comment]
> Dang it, I just realized I didn't sign-off on the last three commits. Is there any sensible way I can fix that, or should I just open a new PR?

I wouldnt bother doing this since it is implied if you are self signing. You are the author and you will be listed in the commit as the author.

I would also just collapse all your commits into one commit using a rebase and force push.

Note when just doing fixups you can also use the following...
````
git add <changed_file>
git add <changed_file2>
.....
git log /* to find the commit you want to apply fixes to */
git commit --fixup commit-id
git push
````
The fixup commits get squashed when they are merged back to master

File: crypto/x509/v3_group_ac.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+{
+    return OPENSSL_strdup("NULL");
+}
+
+static void *s2i_GROUP_AC(const X509V3_EXT_METHOD *method, X509V3_CTX *ctx, const char *str)
+{
+    return ASN1_NULL_new();
+}
+
+/* The groupAC X.509v3 extension is defined in ITU Recommendation X.509
```
Comment (slontis) [inline crypto/x509/v3_group_ac.c:40]: Use the same formatting as the copyright comments above i.e.
````
/*
 * Blah blah
 * Blahdy Blah
 */

**Comment by JonathanWilbur** [comment]
@t8m Is there anything else I need to do to get this PR merged in?

**Comment by JonathanWilbur** [comment]
@paulidale I addressed all feedback on this issue. Is there anything else needed from me?

**Comment by paulidale** [comment]
It needs two reviews and the mandatory waiting period.
Ping @openssl/committers for reviews.

**Review by tmshort — APPROVED** [review]
Lot's of duplication here, but de-duping the code won't necessarily make it any better...

**Comment by JonathanWilbur** [comment]
Sorry Pauli, I didn't know about the waiting period, or I wouldn't have pestered people as much.

And thank you for the review @tmshort . I intentionally had duplicate code to keep with customs established in other similar files, but I would be happy to deduplicate.

**Review by paulidale — APPROVED** [review]

**Comment by tmshort** [comment]
Merged to `master`. Thank you for your contribution!

**Pull request closed without merging**